src/AzureExtension/DevBox/Helpers/WingetConfigWrapper.cs

```diff
@@ -336,13 +381,7 @@ IAsyncOperation<ApplyConfigurationResult> IApplyConfigurationOperation.StartAsyn
             {
                 if (_computeSystemState != ComputeSystemState.Running)
                 {
-                    // Check if the dev box might have been started in the meantime
-                    var stateRequest = await _managementService.HttpsRequestToDataPlane(new Uri(_baseAPI), _devId, HttpMethod.Get, null);
-                    var state = JsonSerializer.Deserialize<DevBoxMachineState>(stateRequest.JsonResponseRoot.ToString(), Constants.JsonOptions)!;
-                    if (state.PowerState != Constants.DevBoxPowerStates.Running)
-                    {
-                        throw new InvalidOperationException(Resources.GetResource(NotRunningFailedKey));
-                    }
+                    await HandleNonRunningState();

```

Comment:
I'm wondering if we could just call the IComputeSystem StartAsync() method instead. If we pass the DevBoxInstance to the WingetConfigWrapper then we'd be able to do that. Then we could use a wait event handle within a while loop that waits for 3 minutes. 

Something like this e.g 
```C#

if (_devBox.GetState() != ComputeSystemState.Running))
{
    var startCancellationTokenSource = new CancellationTokenSource();
    startCancellationTokenSource.CancelAfter(TimeSpan.FromMinutes(30));

    while (_devBoxStartEvent.WaitOne(TimeSpan.FromMinutes(3)))
    {
        if (_devBox.GetState() == ComputeSystemState.Running)
        {
            break;
        }

        if (startCancellationTokenSource.IsCancellationRequested)
        {
            throw new InvalidOperationException(Resources.GetResource(DevBoxErrorStateKey))
        }
 }
```

This would reduce the code amount by a good amount I believe. You can either use a ManualResetEvent or a Task.Delay like you've been using, but you'd have to have an extra variable to keep the loop alive. Also note, we need to increase the time Dev Home's configuration waits too here: https://github.com/microsoft/devhome/blob/ca1ca04bac3325bc68e29f8f478453d2b4f85971/tools/SetupFlow/DevHome.SetupFlow/Models/ConfigureTargetTask.cs#L419

Reply:
+1. One question: would we send _devBoxStartEvent event if the device is already running by the time code in _devBox.StartAsync actually runs (since everything is async and in the cloud the device state can change between our last check and Start command)? This is probably not an issue since we check state every minute which, I assume, would be updated.

Reply:
Good idea; changed.

Reply:
@sshilov7 I just wanted to let you know that your assumption is right. Although that got me thinking of one more edge case:

1. If the `_devBox.StartAsync` method returns an error then we'd be waiting for 15 minutes for no reason. @huzaifa-d, could you add an error check for this one?